src/applications/static-pages/scroll-top-button.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * This button is used for navigating the user back to the top of the page.
+ * It's to be used on long article pages.
+ * There is some complexity in the animation toggling function "scrollListener"
+ * html is located in src/site/components/up_to_top_button.html
+ */
+
+let upToTopButton = '';
+let footer = '';
+let threeFourthsContainer = '';
+const container = document.querySelector('.vsa-top-button-container');
+
+function initVars() {
+  threeFourthsContainer = container.querySelector('.usa-width-three-fourths');
+  upToTopButton = container.querySelector('.vsa-top-button');
+  footer = document.querySelector('.footer-inner');
+  // NOTE: Guarded Return so button does not show at all with errors
+  if (!threeFourthsContainer || !upToTopButton || !footer)
+    return 'NODE_MISSING_CLASS';
+
+  return 'SUCCESS';
+}
+
+function navigateToTop() {
+  return window.scrollTo(0, 0);
+}
+
+function assignClickHandlerButtonToDom() {
+  const button = container.querySelector('.usa-button.vsa-top-button');
+  if (button) button.onclick = navigateToTop();
+  // eslint-disable-next-line no-console
+  if (!button) console.warn('MISSING_UP_TO_TOP_BUTTON');
+}
+
+function scrollListener() {
+  // Responsible for toggling animation classes
+  const scrollFromTop =
+    window.pageYOffset !== undefined
+      ? window.pageYOffset
+      : (document.documentElement || document.body.parentNode || document.body)
+          .scrollTop;
+  const distanceOfScrollingBeforeAppearing = 600;
+  const buttonTransitionIn = 'vsa-top-button-transition-in';
+  const buttonTransitionOut = 'vsa-top-button-transition-out';
+  const buttonContainerRelative = 'vsa-top-button-container-relative';
+  const buttonTransitionReset = 'vsa-top-button-transition-reset';
+
+  function doesElHaveClass(el, className) {
+    return el.classList.value.indexOf(className) > -1;
+  }
+
+  function isScrolledIntoView(el) {
+    const rect = el.getBoundingClientRect();
+    const elemTop = rect.top;
+    // Only partially || completely visible elements return true
+    return elemTop >= 0 && elemTop <= window.innerHeight;
+  }
+
+  if (
+    scrollFromTop > distanceOfScrollingBeforeAppearing &&
+    !doesElHaveClass(upToTopButton, buttonTransitionIn)
+  ) {
+    upToTopButton.classList.add(buttonTransitionIn);
+    upToTopButton.classList.remove(buttonTransitionOut);
+  } else if (
+    scrollFromTop < distanceOfScrollingBeforeAppearing &&
+    doesElHaveClass(upToTopButton, buttonTransitionIn)
+  ) {
+    upToTopButton.classList.add(buttonTransitionOut);
+    upToTopButton.classList.remove(buttonTransitionIn);
+  }
```

Comment:
`classList` also has a method [`toggle`](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle) with an optional boolean arg that can be pretty useful for consolidating if-else blocks that just do add/remove. For example -

```js
const isButtonVisible = scrollFromTop > distanceOfScrollingBeforeAppearing;

upToTopButton.classList.toggle(buttonTransitionIn, isButtonVisible);
```